librz/analysis/esil_dfg.c

```diff
@@ -1066,13 +1026,15 @@ RZ_API RzStrBuf *rz_analysis_esil_dfg_filter(RzAnalysisEsilDFG *dfg, const char
 	// dfs the graph starting at node of esp-register
 	rz_graph_dfs_node_reverse (dfg->flow, resolve_me, &vi);
 
-	RBIter ator;
-	RzAnalysisEsilDFGNode *node;
-	rz_rbtree_cont_foreach (filter.tree, ator, node) {
-		// resolve results to opstr here
-		RzStrBuf *resolved = get_resolved_expr (&filter, node);
-		rz_strbuf_append (filtered, rz_strbuf_get (resolved));
-		rz_strbuf_free (resolved);
+	if (filter.tree->root) {
```

Comment:
this is IMHO wrong. @thestr4ng3r shouldn't we just do this check inside `rz_rbtree_cont_foreach` like we do for lists and others? This is the special case "tree is empty", which I expect to be able to traverse with the regular `rz_rbtree_cont_foreach`.

Reply:
Yes, absolutely. In fact, it probably already works this way.
If we take a look at the foreach:
```c
#define rz_rbtree_cont_foreach(tree, it, dat) \
	for ((it) = rz_rbtree_first (&tree->root->node); ...
```
The dangerous deref would be `tree->root->node` but actually with the `&` it becomes just `(RBNode *)(void *)tree->root` so it's fine unless C specifies the deref to always happen. This could be done in a more expressive way using the `rz_rbtree_iter_get()` macro.